mlflow_elasticsearchstore/elasticsearch_store.py

```diff
@@ -35,21 +37,29 @@ def _hit_to_mlflow_experiment(self, hit: Any) -> Experiment:
                           artifact_location=hit.artifact_location,
                           lifecycle_stage=hit.lifecycle_stage)
 
-    def _hit_to_mlflow_run(self, hit: Any) -> Run:
+    def _hit_to_mlflow_run(self, hit: Any, use_latest_metrics: bool = False) -> Run:
         return Run(run_info=self._hit_to_mlflow_run_info(hit),
-                   run_data=self._hit_to_mlflow_run_data(hit))
+                   run_data=self._hit_to_mlflow_run_data(hit, use_latest_metrics))
 
     def _hit_to_mlflow_run_info(self, hit: Any) -> RunInfo:
         return RunInfo(run_uuid=hit.meta.id, run_id=hit.meta.id,
-                       experiment_id=str(hit.experiment_id), user_id=hit.user_id,
-                       status=hit.status, start_time=hit.start_time,
+                       experiment_id=str(hit.experiment_id),
+                       user_id=hit.user_id if hasattr(hit, 'user_id') else None,
+                       status=hit.status,
+                       start_time=hit.start_time if hasattr(hit, 'start_time') else None,
                        end_time=hit.end_time if hasattr(hit, 'end_time') else None,
                        lifecycle_stage=hit.lifecycle_stage, artifact_uri=hit.artifact_uri)
 
-    def _hit_to_mlflow_run_data(self, hit: Any) -> RunData:
-        return RunData(metrics=[self._hit_to_mlflow_metric(m) for m in hit.metrics],
-                       params=[self._hit_to_mlflow_param(p) for p in hit.params],
-                       tags=[self._hit_to_mlflow_tag(t) for t in hit.tags])
+    def _hit_to_mlflow_run_data(self, hit: Any, use_latest_metrics: bool) -> RunData:
+        return RunData(metrics=[self._hit_to_mlflow_metric(m) for m in
+                                (hit.latest_metrics if hasattr(hit, 'latest_metrics') else [])]
+                       if use_latest_metrics else
+                       [self._hit_to_mlflow_metric(m) for m in
+                        (hit.metrics if hasattr(hit, 'metrics') else [])],
```

Comment:
This is hard to read, please put the if use_last_metrics before.
Also, when do you need to create a RunData object with metrics and not latest_metrics ?